### PR TITLE
refactor(project): replaces lodash with native Array methods

### LINKED ISSUE #693: Replace lodash with ES6 methods and functions
This will allow to use native methods in modern browsers and use polyfills for old one. The main benefit is reduced bundle size and -1 dependency.

**Update**: I can provide a PR if you agree to proceed with this

### PR DESCRIPTION
**Pull request by stalniy**
Fixes #693

### CODE REVIEW

File: demo/knockout2.html [hunk 1]
```diff
@@ -16,7 +16,8 @@
     <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.11.0/jquery-ui.js"></script>
     <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
-    <script src="https://cdnjs.cloudflare.com/ajax/libs/lodash.js/3.5.0/lodash.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/core-js/2.4.1/shim.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/lodash.js/4.17.4/lodash.min.js"></script>
```
Comment (adumesny) [inline demo/knockout2.html]: did you mean to keep lodash in there ?

Comment (stalniy) [inline demo/knockout2.html]: Looks like it was left by accident.

**Comment by stalniy** [comment]
any updates on this?

**Comment by adumesny** [comment]
I think the drivers have fallen off the bandwagon - not seeing any updates nor taking serious bug fixes in for a long time now. That's a problem when you only have 1 or 2 with approval power... what's the point in making fixes if no-one around to accept them ? They should open this up if they don't want to this codebase to die...

**Comment by radiolips** [comment]
Hey @stalniy . Thanks for putting this together! I'm not sure yet that we want to make the leap to ES6. I think we could get rid of lodash and us ES2015 functions, or we could maintain your setup as a separate branch, but I don't think it should be a requirement for gridstack yet. How would you feel about these solutions?

**Comment by creage** [comment]
When could we expect this PR to be merged in? We want as smaller size as possible, and having whole Lodash as dependency for simple operations adds redundant 500+Kbs of it.

We'd like to see either Lodash dropped completely, or, at least, require only [modules you need](https://www.npmjs.com/browse/keyword/lodash-modularized) from Lodash, not whole library, please.

**Comment by radiolips** [comment]
Hey, @creage . As I had mentioned, I want this to be a separate build off of gridstack and not actually part of the master. I'm fine using only the necessary lodash modules if someone wants to put in a PR for that.

**Comment by adumesny** [comment]
@stalniy how do I merge your changes locally to take your review ?
I tried: 'git pull origin pull/703/head'
but I get: fatal: Couldn't find remote ref pull/703/head

**Comment by stalniy** [comment]
@adumesny just clone my gridstack repo and `git checkout refactor/lodash`

**Comment by adumesny** [comment]
@stalniy I ended up using the main branch and `git pull origin pull/703/head` to merge (and fix conflict) your changes over! thank you for doing that work. Using it locally...

**Comment by crazyfx1** [comment]
Can this be accepted and released as v0.3.1?

**Comment by radiolips** [comment]
@adumesny did this work okay when you used it? @stalniy can you update the PR to fix the conflicts? I'd be happy to merge this in.

**Comment by adumesny** [comment]
@radiolips yes it works great, been using it for couple months now. the main conflict were the removal of the dist/ content (trivial) and some small code changes since the original PR was made. I'm not sure how to modify his PR to give him credit for his work, otherwise I would be happy to fix the conflicts...

**Comment by stalniy** [comment]
Hi guys,

I’m currently on vacation and won’t be able to work on this till the next Thursday. So, go ahead!

**Comment by radiolips** [comment]
@stalniy are you sure? It's already been open nearly a year (sorry), it can wait one additional week if you want to handle it. Also, have a great vacation!

**Comment by radiolips** [comment]
@adumesny Do you still have time to update and re-submit this PR with the conflicts removed? If so, I'd be happy to get this merged in.

**Comment by adumesny** [comment]
let's wait for @stalniy to update his...

**Comment by stalniy** [comment]
@adumesny , @radiolips I merged develop branch. All tests passed. Please check whether everthing is ok

**Comment by stalniy** [comment]
For all IE. The rest supports this funxtion. The full compatibility https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

There are some `find` occurrences which can be replaced with `some` (some is supported from IE9) but I didn’t want to force people to include 2 polyfills for IE8 for example 

That’s why used `Boolean(array.find(...))`

**Comment by radiolips** [comment]
@stalniy I hate to do this to you, but could you update your PR one-more-time? I updated lodash in the last release, which required me to update several of the demos, config files, and readmes. Could you update them again, and I'll merge this in?

**Comment by adumesny** [comment]
would that be in 0.4.1 then ? I see you released 0.4 recently. thanks.

**Comment by stalniy** [comment]
@radiolips I won’t. You can do it yourself if you think this PR is important.

**Pull request merged**